src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -664,11 +664,11 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
                 << deleteBatch << " rows.";
             long long totalRowsAffected = 0;
             long long rowsAffected;
-            soci::statement st = [&]
+            boost::optional<soci::statement> st;
             {
                 auto db = ledgerDb_->checkoutDb();
-                return (db->prepare << deleteQuery);
-            }();
+                st.emplace(db->prepare << deleteQuery);
```

Comment:
This is all true, but the `boost::optional` removes the need to rely on anything about the implementation of `statement`. I don't really see `boost::optional` as all that exotic.